src/parser/esprima_cpp/esprima.cpp

```diff
@@ -2752,9 +2756,11 @@ class Parser {
                 bool previousAllowYield = this->context->allowYield;
                 bool previousAwait = this->context->await;
                 bool previousInArrowFunction = this->context->inArrowFunction;
+                bool previousAllowStrictDirective = this->context->allowStricDirective;
 
                 this->context->allowYield = false;
                 this->context->inArrowFunction = true;
+                this->context->allowStricDirective = !this->currentScopeContext->m_hasParameterOtherThanIdentifier;
```

Comment:
This line should be executed after `extractNamesFromFunctionParams(list)` because `m_hasParameterOtherThanIdentifier` is calculated in `extractNamesFromFunctionParams`.